Adding DevTools docs and updating tooling pages

sfshaza2 opened the pull request:
This is a large PR. Here is a list of (some of the) staged relevant pages:

https://sz-flutter.firebaseapp.com/docs/development/tools/devtools/overview
  and all of the pages below /devtools

https://sz-flutter.firebaseapp.com/docs/testing/debugging
https://sz-flutter.firebaseapp.com/docs/testing/code-debugging
https://sz-flutter.firebaseapp.com/docs/testing/ui-performance
https://sz-flutter.firebaseapp.com/docs/testing/best-practices

I’ve put Kenzie and Jacob as reviewers, but please add anyone else that I overlooked.

Also see:  @devoncarew

jacob314 commented on src/_data/sidenav.yml:
```diff
@@ -142,8 +142,27 @@
+          children:
+            - title: Overview
+              permalink: /docs/development/tools/devtools/overview
+            - title: From Android Studio
```
nit: should we have a subheading of "installing" that contains these three "from" items?

sfshaza2 replied:
The reason I didn’t do that was because we can’t go another level with our side nav. We are limited to this number of levels.

jacob314 commented on src/docs/development/tools/android-studio.md:
```diff
@@ -87,88 +87,78 @@ The Dart plugin performs code analysis that enables:
-start the app in **Debug** mode, and then open the Performance tool window
-using **View > Tool Windows > Flutter Performance**.
+{{site.alert.note}}
+  The Performance pane is not yet available in DevTools,
```
This doesn't seem right. We should direct users to the devtools timeline page if they want to understand performance. The IntelliJ performance pane just directs users to that page to see more information.

sfshaza2 replied:
ok!

jacob314 commented on src/docs/development/tools/vs-code.md:
```diff
@@ -118,102 +142,95 @@ of your changes almost instantly with the _Stateful Hot Reload_ feature. See
-* **Toggle Repaint Rainbow**: Show rotating colors on layers when repainting.
-* **Toggle Slow Animations**: Slow down animations to enable visual inspection.
-* **Toggle Slow-Mode Banner**: Hide the slow mode banner even when running a
+* **Toggle Baseline Painting**: Causes each RenderBox to paint a line
```
Might mention these can also be toggled via buttons in the inspector.
Bonus points: include the icons from the inspector next to each icon in this list to use a consistent vocabulary.

kenzieschmoll commented on src/docs/development/tools/devtools/overview.md:
```diff
@@ -0,0 +1,50 @@
+a Flutter app, diagnosing UI jank performance issues in
+a Flutter app, source-level debugging of a Flutter or
+Dart command-line app, and viewing general log and
+diagnostics information about a running app.
```
maybe say something in here about diagnosing memory issues as well

kenzieschmoll commented on src/docs/development/tools/android-studio.md:
```diff
@@ -87,88 +87,78 @@ The Dart plugin performs code analysis that enables:
+
+  The instructions below describe features available in Android
+  Studio and IntelliJ. For information on launching DevTools,
+  see [Running DevTools from Android Studio][] in the
```
Should all references to "from Android Studio" actually say "from Android Studio / IntelliJ"? It seems like we mention both IDEs elsewhere in the docs (Ex. Tools & techniques -> Android Studio / IntelliJ)

sfshaza2 commented:
Hi @jacob314 and @kenzieschmoll!  I’m sorry this PR is so unwieldy, but thanks for looking at it anyway.

I fixed the issues you pointed out, including adding images for the various inspector icons. I also fixed some broken links/anchors I had missed earlier.

The devtools front page in this PR is not much different than the wiki version, except I added the image and GIF files. In general, as I migrated the files over, I changed some of the text to match our conventions and advice in the Google Style Guide: https://developers.google.com/style/

Thanks for taking another look!

kenzieschmoll commented on src/_data/sidenav.yml:
```diff
@@ -142,8 +142,27 @@
+          children:
+            - title: Overview
+              permalink: /docs/development/tools/devtools/overview
+            - title: Install from Android Studio
```
same comment as before regarding Android Studio vs Android Studio / IntelliJ

sfshaza2 replied:
Initially, I was a little leery of bloating the side nav, but I went ahead and added it.  :)

sfshaza2 commented:
Thanks, @kenzieschmoll! I’ve updated the PR. @jacob314, did you plan to review again?

sfshaza2 commented:
I’m going to land this. Please file issues or PRs if you find any problems.
